diff --git a/package/libopenssl/libopenssl.mk b/package/libopenssl/libopenssl.mk
index ae6658ed40..35164bb875 100644
--- a/package/libopenssl/libopenssl.mk
+++ b/package/libopenssl/libopenssl.mk
@@ -5,8 +5,8 @@
 ################################################################################
 
 LIBOPENSSL_VERSION = 1.1.1t
-LIBOPENSSL_SITE = https://www.openssl.org/source
-LIBOPENSSL_SOURCE = openssl-$(LIBOPENSSL_VERSION).tar.gz
+LIBOPENSSL_SITE = $(ET_OPENSSL_SOFTWARE_DIR)
+LIBOPENSSL_SITE_METHOD = local
 LIBOPENSSL_LICENSE = OpenSSL or SSLeay
 LIBOPENSSL_LICENSE_FILES = LICENSE
 LIBOPENSSL_INSTALL_STAGING = YES
